src/Analyzer/JoinNode.h

```diff
@@ -163,5 +163,57 @@ class JoinNode final : public IQueryTreeNode
     static constexpr size_t children_size = join_expression_child_index + 1;
 };
 
+class CrossJoinNode;
+using CrossJoinNodePtr = std::shared_ptr<CrossJoinNode>;
+
+class CrossJoinNode final : public IQueryTreeNode
+{
+public:
+    /** Construct join node with left table expression, right table expression and join expression.
+      * Example: SELECT id FROM test_table_1 INNER JOIN test_table_2 ON expression.
+      *
+      * test_table_1 - left table expression.
+      * test_table_2 - right table expression.
+      * expression - join expression.
+      */
+    explicit CrossJoinNode(QueryTreeNodes table_expressions);
+
+    const QueryTreeNodes & getTableExpressions() const
+    {
+        return children;
+    }
+
+    QueryTreeNodes & getTableExpressions()
+    {
+        return children;
```

Comment:
I'd made this node a single child with the `ListNode` type. imo, it's a bit easier to understand and maintain, but not significantly so.